### PR TITLE
add a forward constructor in PriorityQueue to HeapPriorityQueue

### LINKED ISSUE #17: PriorityQueue should forward a constructor to HeapPriorityQueue
It's nice that `PriorityQueue` allows for different implementations, but (like the core collection APIs) it should have a factory constructor that forwards to the default implementation. Right now, it's confusing that you can't construct a `PriorityQueue` using `new PriorityQueue()`.

### PR DESCRIPTION
**Pull request by a14n**
Fixes #17

### CODE REVIEW

File: lib/priority_queue.dart [hunk 1]
```diff
@@ -13,6 +13,16 @@ import "dart:collection" show SplayTreeSet;
...
 abstract class PriorityQueue<E> {
   /**
+   * Creates an empty [PriorityQueue].
+   *
+   * The created [PriorityQueue] is a plain [HeapPriorityQueue]. As such,
+   * [Comparable.compare] is used to compare the priority of elements.
+   *
+   * The queue is equivalent to one created by `new HeapPriorityQueue<E>()`.
+   */
+  factory PriorityQueue() = HeapPriorityQueue<E>;
```
Comment (nex3) [inline lib/priority_queue.dart]: It would be good to give this a comparator argument as well.

Comment (a14n) [inline lib/priority_queue.dart]: I used the same pattern as `Set()` that forward to `LinkedHashSet` (https://github.com/dart-lang/sdk/blob/master/sdk/lib/core/set.dart#L45 and https://github.com/dart-lang/sdk/blob/master/sdk/lib/collection/linked_hash_set.dart#L77). Let me know what you prefer.

Comment (nex3) [inline lib/priority_queue.dart]: I'm not really sure why the default Set constructor doesn't have those arguments, actually. Regardless, the ordering of a priority queue is pretty central to its behavior, so it should be customizable in the main constructor.

Comment (a14n) [inline lib/priority_queue.dart]: done (commit amended)

**Pull request merged**